Xamarin.Forms.Platform.Android/ImageCache.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	/// <summary>
+	/// I setup the access to all the cache elements to be async because
+	/// if I didn't then it was locking up the GC and freezing the entire app
+	/// </summary>
+	class ImageCache
+	{
+		readonly FormsLruCache _lruCache;
+		readonly ConcurrentDictionary<string, SemaphoreSlim> _waiting;
+
+		public ImageCache() : base()
+		{
+			_waiting = new ConcurrentDictionary<string, SemaphoreSlim>();
+			_lruCache = new FormsLruCache();			
+		}
+
+		async Task PutAsync(string key, TimeSpan cacheValidity, global::Android.Graphics.Bitmap cacheObject)
+		{
+			await Task.Run(() =>
```

Comment:
Since PutAsync is only called from GetAsync, it's already running on a thread from the thread pool; is there any reason to use Task.Run again?

Reply:
This was a bit of an overreaction. When I did all the cache interactions synchronously it started locking up the GC so I just moved everything to tasks to see if that would unblock that issue which it did. I've changed put to now just be synchronous and everything seems to run well.

Issue4915 is a good test for this one